crates/rpc/src/handlers/state.rs

```diff
@@ -65,22 +85,90 @@ pub async fn get_state_from_id(state_id: ID, db: &ReamDB) -> Result<BeaconState,
         )))
 }
 
-pub async fn get_state(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}")]
+pub async fn get_beacon_state(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    let state_id = ID::from_str(&state_id.into_inner()).map_err(error::ErrorBadRequest)?;
+    let state = get_state_from_id(state_id, &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
-    Ok(with_status(BeaconResponse::json(state), StatusCode::OK))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state)))
 }
 
-pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}/root")]
+pub async fn get_state_root(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    let state_id = ID::from_str(&state_id.into_inner()).map_err(error::ErrorBadRequest)?;
+    let state = get_state_from_id(state_id, &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
     let state_root = state.tree_hash_root();
 
-    Ok(with_status(
-        BeaconResponse::json(RootResponse::new(state_root)),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(BeaconResponse::from(state_root.to_string())))
 }
+
+/// Called by `/eth/v1/beacon/states/{state_id}/fork` to get fork of state.
+#[get("/beacon/states/{state_id}/fork")]
+pub async fn get_state_fork(
+    db: web::Data<ReamDB>,
+    state_id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    let state_id = ID::from_str(&state_id.into_inner()).map_err(error::ErrorBadRequest)?;
+    let state = get_state_from_id(state_id, &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
+
+    Ok(HttpResponse::Ok().json(state.fork))
+}
```

Comment:
```suggestion
/// Called by `/eth/v1/beacon/states/{state_id}/fork` to get fork of state.
#[get("/beacon/states/{state_id}/fork")]
pub async fn get_state_fork(
    db: web::Data<ReamDB>,
    state_id: web::Path<ID>,
) -> actix_web::Result<impl Responder> {
    let state = get_state_from_id(state_id, &db).await?;
    Ok(HttpResponse::Ok().json(BeaconResponse::new(state.fork)))
}
```

This function should look like this